libraries/abstractions/common_io/test/test_iot_flash.c

```diff
@@ -488,7 +488,8 @@ TEST( TEST_IOT_FLASH, AFQP_IotFlashEraseFlashBlocks )
         TEST_ASSERT_NOT_EQUAL( NULL, pxFlashInfo );
 
         /* Make sure the offset is aligned to BlockSize */
-        if( ultestIotFlashStartOffset == ( ultestIotFlashStartOffset & pxFlashInfo->ulBlockSize ) )
+        if( ( ultestIotFlashStartOffset == ( ultestIotFlashStartOffset & pxFlashInfo->ulBlockSize ) ) || 
```

Comment:
Assuming `pxFlashInfo->ulBlockSize` is a power of 2 could these two checks can be combined to a single check as `( ultestIotFlashStartOffset & ( pxFlashInfo->ulBlockSize - 1 ) ) ! = 0` ?

You can additionally have a check for block size is power of 2 by `( ( pxFlashInfo->ulBlockSize != 0 ) && ( ( pxFlashInfo->ulBlockSize & ( pxFlashInfo->ulBlockSize - 1 ) ) == 0)`

Reply:
Thank you. I agree with your opinion. Code is updated.

Reply:
Thanks, It may makes sense to move the logic of checking if blocksize is a power of 2  into a separate `if` condition with a TEST assert with relevant message. Or atleast, could we leave a comment so its clear what is being done.

Reply:
Thanks. 
I think the better location of checking if block size is power of 2 is where validating the flash info. 
And I also checked the flash size, block size, sector size and page size.